tests/src/Kernel/ListBuilderTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_list_pages\Kernel;
+
+use Drupal\oe_list_pages\ListBuilder;
+use Drupal\oe_list_pages\ListPageConfiguration;
+use Drupal\oe_list_pages\ListSourceFactory;
+
+/**
+ * Tests the list builder.
+ */
+class ListBuilderTest extends ListsSourceTestBase {
+
+  /**
+   * Tests the facet cache tags are correctly applied on the render arrays.
+   */
+  public function testFacetCacheTags(): void {
+    $default_list_id = ListSourceFactory::generateFacetSourcePluginId('entity_test_mulrev_changed', 'item');
+    $this->facet = $this->createFacet('created', $default_list_id, '', 'oe_list_pages_multiselect', []);
+    $this->facet->addProcessor([
+      'processor_id' => 'oe_list_pages_date_status_processor',
+      'weights' => [],
+      'settings' => [],
+    ]);
+    $this->facet->save();
+
+    $configuration = [
+      'entity_type' => 'entity_test_mulrev_changed',
+      'bundle' => 'item',
+      'exposed_filters' => [],
+      'exposed_filters_overridden' => FALSE,
+      'default_filter_values' => [],
+      'contextual_filters' => [],
+    ];
+    $listPageConfiguration = new ListPageConfiguration($configuration);
+
+    $listExecutionManager = $this->container->get('oe_list_pages.execution_manager');
+    $pager = $this->container->get('pager.manager');
+    $entityRepository = $this->container->get('entity.repository');
+    $formBuilder = $this->container->get('form_builder');
+    $facetsUrlGenerator = $this->container->get('facets.utility.url_generator');
+    $processorManager = $this->container->get('plugin.manager.facets.processor');
+    $requestStack = $this->container->get('request_stack');
+    $urlProcessorManager = $this->container->get('plugin.manager.facets.url_processor');
+    $multiselectFilterManager = $this->container->get('plugin.manager.multiselect_filter_field');
+    $listSourceFactory = $this->container->get('oe_list_pages.list_source.factory');
+
+    $listBuilder = new ListBuilder($listExecutionManager, $this->entityTypeManager, $pager, $entityRepository, $formBuilder, $facetsUrlGenerator, $processorManager, $requestStack, $urlProcessorManager, $multiselectFilterManager, $listSourceFactory);
```

Comment:
$listBuilder = $this->container->get('oe_list_pages.builder'); 

All the remaining services don't need to be instantiated.